### DISCUSSION TITLE
uc-crux-llvm

### DISCUSSION THREAD

**Discussion by cortezcodes** [GaloisInc/crucible#1132] [Q&A]
I am currently working on a research project that I am trying to use uc-crux-llvm to test some open source codebases for bugs. I was wondering if there is any functionality built into uc-crux-llvm that would help with compiling multiple file codebases into one LLVM bitcode module? I am attempting to run some analysis on OpenSSL's famous Heartbleed vulnerability. Thanks for the help!

**Comment by kquick** [comment]
The build-bom tool can be useful to obtain LLVM BC from a multi-file build process.  See https://github.com/travitch/build-bom

**Reply by langston-barrett** [reply]
build-bom is a great tool, [gllvm](https://github.com/SRI-CSL/gllvm) is another possibility.

**Reply by cortezcodes** [reply]
Awesome! Thank you! I will check these out and see if I can get it to work for my use case.
